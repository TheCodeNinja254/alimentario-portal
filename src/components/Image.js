import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import makeStyles from "@material-ui/core/styles/makeStyles";

const useStyles = makeStyles(() => ({
  root: {
    objectFit: "cover",
    maxWidth: "100%",
  },
}));

const Image = React.forwardRef(({ src, alt, className, ...rest }, ref) => {
  const classes = useStyles();
  return (
    <img
      src={src}
      alt={alt}
      className={clsx(className, classes.root)}
      ref={ref}
      {...rest}
    />
  );
});

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string,
};

export default React.memo(Image);
