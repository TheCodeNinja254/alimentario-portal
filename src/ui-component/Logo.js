import React from "react";

import { Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import logo from "../assets/images/alimentarioLogo.svg";

// ===========================|| LOGO SVG ||=========================== //
const useStyles = makeStyles((theme) => ({
  logo: {
    marginRight: theme.spacing(3),
  },
}));

const Logo = () => {
  const classes = useStyles();

  return (
    <>
      <img
        src={logo}
        className={classes.logo}
        alt="Desafio Alimentario"
        width="50"
        height="50"
      />
      <Typography variant="h3" color="secondary">
        Desafio Alimentario
      </Typography>
    </>
  );
};

export default Logo;
