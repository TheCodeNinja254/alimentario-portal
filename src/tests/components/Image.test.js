import React from "react";
import renderer from "react-test-renderer";
import Logo from "../../../public/favicons/saf-logo.png";
import Image from "../../components/Image";

describe("<Image />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Image src={Logo} alt="Logo" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
