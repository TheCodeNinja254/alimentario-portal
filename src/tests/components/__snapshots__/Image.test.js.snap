// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders correctly 1`] = `
<img
  alt="Logo"
  className="makeStyles-root-1"
  src="saf-logo.png"
/>
`;
